File: problem/Water_Bottles/numWaterBottles.js
/**
 * @param {number} numBottles
 * @param {number} numExchange
 * @return {number}
 */
var numWaterBottles = function(numBottles, numExchange) {
    let drinked = 0, eb = 0;
    while (numBottles > 0) {
        drinked += numBottles;
        eb += numBottles;
        numBottles = Math.floor(eb / numExchange);
        eb = eb % numExchange;
    }
    return drinked;
};

File: problem/Water_Bottles/numWaterBottles.cpp
class Solution
{
public:
    int numWaterBottles(int numBottles, int numExchange)
    {
        int drinked = 0, eb = 0;
        while (numBottles > 0)
        {
            drinked += numBottles;
            eb += numBottles;
            numBottles = eb / numExchange;
            eb = eb % numExchange;
        }
        return drinked;
    }
};